### ISSUE TITLE
Allow load_data to be templated

### ISSUE THREAD

**Issue by mcexit** [getzola/zola#2018]
I'm pretty new to Tera and Zola, but I would like to use a data file with variables. Maybe there is another way to do this already and I'm just not seeing it. Another option would be to allow the `set` function to create complex objects and set nested values, such as:

```
{% set data = load_data(path="data/index.toml") %}
{% set data.title = page.title %}
```

**Comment by timharek** [comment]
This is already possible, but you need to place your data-files either within the `content`- or `static`-folder, then the file will be reachable.

You can read more about it here:
https://www.getzola.org/documentation/templates/overview/#load-data

**Comment by mcexit** [comment]
Yes I'm aware that this method already exists, but those data files are not templated. There is no filter to process them through the template engine either. There needs to be a way to include variables inside them and have them processed:

`index.toml`
```
'@context' = 'https://schema.org'
'@type' = 'WebSite'
title = '{{ section.title }}'
```

**Comment by Keats** [comment]
Passing files through Tera is not going to happen though, it seems very niche
